encoding/kzg/pointsIO.go

```diff
@@ -191,207 +180,63 @@ func ReadG1PointSection(filepath string, from, to uint64, numWorker uint64) ([]b
-		return nil, err
-	}
-
-	defer func() {
-		if err := g1f.Close(); err != nil {
-			log.Printf("g2 close error %v\n", err)
-		}
-	}()
-
-	startTimer := time.Now()
-	g1r := bufio.NewReaderSize(g1f, int(n*G2PointBytes))
-
-	if n < numWorker {
-		numWorker = n
-	}
-
-	buf, err := ReadDesiredBytes(g1r, n*G2PointBytes)
-	if err != nil {
-		return nil, err
-	}
-
-	// measure reading time
-	t := time.Now()
-	elapsed := t.Sub(startTimer)
-	log.Printf("    Reading G2 points (%v bytes) takes %v\n", (n * G2PointBytes), elapsed)
-
-	startTimer = time.Now()
-
-	s2Outs := make([]bn254.G2Affine, n)
-
-	results := make(chan error, numWorker)
-
-	start := uint64(0)
-	end := uint64(0)
 	size := n / numWorker
-	for i := uint64(0); i < numWorker; i++ {
-		start = i * size
 
+	for i := uint64(0); i < numWorker; i++ {
+		start := i * size
+		end := start + size
 		if i == numWorker-1 {
 			end = n
-		} else {
-			end = (i + 1) * size
 		}
 
-		go readG2Worker(buf, s2Outs, start, end, G2PointBytes, results)
+		go func(start, end uint64) {
+			for j := start; j < end; j++ {
+				pointData := buf[j*pointSizeBytes : (j+1)*pointSizeBytes]
+				switch p := any(&points[j]).(type) {
+				case *bn254.G1Affine:
+					if _, err := p.SetBytes(pointData); err != nil {
+						results <- fmt.Errorf("error setting G1 point bytes: %w", err)
+						return
+					}
+				case *bn254.G2Affine:
+					// G2 points are stored as uncompressed points, so we can directly set bytes
+					if _, err := p.SetBytes(pointData); err != nil {
+						results <- fmt.Errorf("error setting G2 point bytes: %w", err)
+						return
+					}
+				default:
+					results <- fmt.Errorf("unsupported point type: %T", p)
+					return
+				}
+			}
+			results <- nil
+		}(start, end)
 	}
 
 	for w := uint64(0); w < numWorker; w++ {
-		err := <-results
-		if err != nil {
+		if err := <-results; err != nil {
 			return nil, err
 		}
 	}
 
-	// measure parsing time
-	t = time.Now()
-	elapsed = t.Sub(startTimer)
-	log.Println("    Parsing takes", elapsed)
-
-	return s2Outs, nil
+	return points, nil
 }
 
-// from is inclusive, to is exclusive
-func ReadG2PointSection(filepath string, from, to uint64, numWorker uint64) ([]bn254.G2Affine, error) {
-	if to <= from {
-		return nil, fmt.Errorf("the range to read is invalid, from: %v, to: %v", from, to)
-	}
-	g2f, err := os.Open(filepath)
-	if err != nil {
-		log.Println("ReadG2PointSection.ERR.0", err)
-		return nil, err
-	}
-
-	defer func() {
-		if err := g2f.Close(); err != nil {
-			log.Printf("error %v\n", err)
-		}
-	}()
-
-	n := to - from
-
-	g2r := bufio.NewReaderSize(g2f, int(n*G2PointBytes))
-
-	_, err = g2f.Seek(int64(from*G2PointBytes), 0)
-	if err != nil {
-		return nil, err
-	}
-
-	if n < numWorker {
-		numWorker = n
-	}
-
-	buf, err := ReadDesiredBytes(g2r, n*G2PointBytes)
+// readDesiredBytes reads exactly numBytesToRead bytes from the reader and returns
+// the result.
+func readDesiredBytes(reader *bufio.Reader, numBytesToRead uint64) ([]byte, error) {
```

Comment:
Why `readDesiredBytes` instead of just `readBytes`? I think `desired` is implied

Reply:
Fixed in commit c3d0462d - renamed readDesiredBytes to readBytes